src/Sema.zig

```diff
@@ -21751,7 +21745,10 @@ fn reifyStruct(
         if (field_is_comptime) any_comptime_fields = true;
         if (field_default_value != .none) any_default_inits = true;
         switch (try field_alignment_val.orderAgainstZeroSema(pt)) {
-            .eq => {},
+            .eq => if (layout != .@"packed") {
+                // this is an invalid case, however we want to error after this loop
```

Comment:
Looks to me like we should just remove `any_aligned_fields` from this function; it's now always `true` in every non-trivial case.